lightning/src/onion_message/functional_tests.rs

```diff
@@ -196,14 +212,24 @@ fn create_nodes_using_cfgs(cfgs: Vec<MessengerCfg>) -> Vec<MessengerNode> {
 		);
 		let offers_message_handler = Arc::new(TestOffersMessageHandler {});
 		let custom_message_handler = Arc::new(TestCustomMessageHandler::new());
+		let messenger = if cfg.intercept_offline_peer_oms {
+			OnionMessenger::new_with_offline_peer_interception(
+				entropy_source.clone(), node_signer.clone(), logger.clone(),
+				node_id_lookup, message_router, offers_message_handler,
+				custom_message_handler.clone()
+			)
+		} else {
+			OnionMessenger::new(
+				entropy_source.clone(), node_signer.clone(), logger.clone(),
+				node_id_lookup, message_router, offers_message_handler,
+				custom_message_handler.clone()
+			)
+		};
```

Comment:
Instead of having separate constructors, consider a two-variant enum for the offline forwarding strategy. Then that  can be used in other places instead of a bool.

Reply:
Do you mean having only one public constructor for the messenger that takes the enum? I kinda wanted to "hide" the interception API and make it clear what the default constructor should be, since interception is for such a specific use case.

Reply:
Yeah, though either is fine by me.